adding show/hide button

istro opened the pull request:
Added a button to show/hide "complete" todos.

Also made a draft for a pre-commit hook to run linters locally.

istro committed:
add button, but no functionality yet

golmansax commented on app/assets/javascripts/components/todo_list_container.js.jsx:
```diff
@@ -7,12 +7,36 @@ Chk.TodoListContainer = (function () {
       }
     },
 
+    _toggleComplete: function(event) {
```
linter wants `function (`

golmansax replied:
also `event` is defined and not used

golmansax commented on app/assets/javascripts/components/todo_list_container.js.jsx:
```diff
@@ -7,12 +7,36 @@ Chk.TodoListContainer = (function () {
+    _toggleComplete: function(event) {
+      var newState = this.state.show_all ? false : true;
+      this.setState({
+        'show_all': newState
```
linter wants `show_all: newState` (no quotes)

golmansax commented on app/assets/javascripts/components/todo_list_container.js.jsx:
```diff
@@ -7,12 +7,36 @@ Chk.TodoListContainer = (function () {
 
+    getInitialState: function () {
+      return {
+        show_all: true,
```
linter wants no comma after (it breaks old IE browsers)

golmansax commented on app/assets/javascripts/components/todo_list_container.js.jsx:
```diff
@@ -7,12 +7,36 @@ Chk.TodoListContainer = (function () {
+    getInitialState: function () {
+      return {
+        show_all: true,
+      }
```
linter wants semicolon here

istro committed:
Make lint happy, add icon, add pre-commit hook draft

Fixed errors that made build broken (i think), added
a glyph to the icon (so that I would have some css
that would get linted), made a draft for a pre-commit
hook to run linter locally.

istro committed:
Pre-commit linter - make rubocop happy

also add method for rubocop linting

istro committed:
shorter lines! happy now, lint?

istro committed:
improve comment of pre-commit hook script
